parse-package-set/src/Main.purs

```diff
@@ -49,13 +54,16 @@ main :: Effect Unit
 main = runAff_ (either throwException mempty) do
   tmpPath <- liftEffect $ tmpdir "cst-integration-"
 
-  writeTextFile UTF8 (tmpPath <> "/spago.dhall") defaultSpagoDhall
+  liftEffect $ Console.log $ "Making new project in " <> tmpPath
 
-  let execOpts = Exec.defaultExecSyncOptions { cwd = Just tmpPath }
-  s <- liftEffect $ Buffer.toString UTF8 =<< Exec.execSync "spago ls packages" execOpts
-  let lines = Str.split (Str.Pattern "\n") s
-  let packages = Str.joinWith " " (String.takeWhile (_ /= ' ') <$> lines)
-  _ <- liftEffect $ Exec.execSync ("spago install " <> packages) execOpts
+  writeTextFile UTF8 (tmpPath <> "/spago.yaml") defaultSpagoYaml
+
+  let execOptsFn execSyncOptions = execSyncOptions { cwd = Just tmpPath }
+  s <- liftEffect $ Buffer.toString UTF8 =<< Exec.execSync' "spago ls packages --json" execOptsFn
+  let (objectOrError :: Either JsonDecodeError (Object Json)) = Data.Argonaut.Decode.decodeJson =<< Data.Argonaut.Decode.parseJson s
+  (object :: Object Json) <- either (throwError <<< error <<< Data.Argonaut.Decode.printJsonDecodeError) pure objectOrError
+  let (packages :: Array String) = Foreign.Object.keys object
```

Comment:
Are all these types and bindings necessary? Maybe:
```suggestion
  packages <- either (throwError <<< error <<< printJsonDecodeError) (pure <<< Object.keys) $ decodeJson =<< parseJson s
```
Or avoid `either` if you don't like the way it looks:
```suggestion
packages <- case decodeJson =<< parseJson s of
  Left err -> throwError $ error $ printJsonDecodeError err
  Right obj -> pure $ Object.keys obj
```